File: 20211002/B.py
a = input()
b = input()
printer = "No"

if not a == b:
    bool_list = [x for x in range(len(a))]
    false_count = 0
    for i in range(len(a)):
        if(a[i] == b[i]):
            bool_list[i] = True
        else:
            bool_list[i] = False
            false_count+=1

    if false_count != 2:
        printer = "No"
    else:
        for i in range(len(a)):
            if (not bool_list[i]) and not bool_list[i+1]:
                if(a[i] == b[i+1] and b[i] == a[i+1]):
                    printer = "Yes"
                    break
else :
    printer = "Yes"
print(printer)

File: 20211002/b.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    string a,b;
    cin >> a >> b ;

    bool array[a.length()];
    int false_count = 0;

    for(int i=0;i<a.length();i++){
        if (a[i] == b[i]){
            array[i] = true;
        } else {
            array[i] = false;
            false_count++;
        }
    }

    if (false_count != 2){
        cout << "No" << endl;
        return 0;
    }

    for (int i = 0 ;i<a.length();i++){
        if(array[i]){
            continue;
        }
        if(!array[i+1]){
            cout << "Yes" << endl;
            return 0;
        }
    }

    return 0;
}
